jupyter_console/interactiveshell.py

```diff
@@ -597,3 +637,21 @@ def init_history(self):
         self.history_manager = ZMQHistoryManager(client=self.client)
         self.configurables.append(self.history_manager)
 
+    def get_source_and_reset(self):
```

Comment:
I still think it would be more readable inline (the performance difference is trivial, I'm not trying to optimise it), but I'm happy enough to give you the author's privilege to determine matters of style.